### ISSUE TITLE
Inquiry About MAE Performance on PURE Dataset with Default Parameters

### ISSUE THREAD

**Issue by Junjie31** [zizheng-guo/RhythmMamba#9]
Hello,

I am currently working with the PURE dataset using the default parameters, and I have observed that the mean absolute error (MAE) is 0.27. I wanted to ask if this level of error is considered acceptable. The paper mentions an MAE of 0.23, so I am curious about the discrepancy.

Additionally, are there any other parameters or settings I can adjust to improve the model's performance and achieve better results? I also noted that the loss at the last epoch was 4.63.

Any insights or recommendations would be greatly appreciated!

Thank you!

**Comment by zizheng-guo** [comment]
This might be within the error range introduced by initialization. In fact, I have tested this on five different servers, and the results were 0.19 for one run, 0.23 for two runs, and 0.27 for two runs. Additionally, the results from each of the 30 epochs during my training might be helpful for you.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/guoguo/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/guoguo/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">


Epoch | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
MAE | 27.16 | 25.85 | 24.35 | 10.22 | 7.01 | 0.51 | 8.73 | 0.29 | 1.77 | 0.31 | 5.46 | 0.27 | 0.33 | 0.22 | 0.29
Epoch | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 | 25 | 26 | 27 | 28 | 29
MAE | 8.88 | 0.24 | 0.23 | 0.23 | 0.23 | 0.23 | 0.16 | 0.13 | 0.25 | 0.25 | 0.25 | 0.27 | 0.23 | 0.27 | 0.23



</body>

</html>

Since there is no validation set, the result from the final epoch was used as the outcome.
It can be observed that around the 20th epoch, the result stabilized at 0.23, with the optimal value reaching as low as 0.13. Many previous methods trained for 20 or 25 epochs, but to remain consistent with all other tests, we trained for 30 epochs.

**Comment by Junjie31** [comment]
Thank you for your response! I find the initialization perturbation a bit surprising, as it seems quite large. Are there any strategies to control such errors?

I also appreciate you providing some data points. Thank you again for your help!
